### PR TITLE
Backport move(Emplace) fixes from Phobos

### PR DESCRIPTION
**Pull request by MoonlightSentinel**

### CODE REVIEW

File: src/core/lifetime.d [hunk 1]
```diff
@@ -1793,18 +1794,23 @@ void moveEmplace(T)(ref T source, ref T target) @system
...
+        assert((() @trusted => &source !is &target)(), "source and target must not be identical");
 
         static if (hasElaborateAssign!T || !isAssignable!T)
-            memcpy(&target, &source, T.sizeof);
+            () @trusted { memcpy(&target, &source, T.sizeof); }();
         else
             target = source;
 
+        static if (hasElaborateMove!T)
+            __move_post_blt(target, source);
```
[the remark this answers is not in the document]
Comment (MoonlightSentinel) [reply src/core/lifetime.d:1812]: It certainly needs more work but the druntime/phobos versions of these functions shouldn't differ.

**Comment by 12345swordy** [comment]
cc @thewilsonator

**Comment by radcapricorn** [comment]
I'd rather if this was looked over wrt __move_post_blt and the de-facto status of DIP1014.

File: src/core/lifetime.d (hunk 1 above)
[the remark this answers is not in the document]
Comment (radcapricorn) [reply src/core/lifetime.d:1812]: I'll just ping @RazvanN7 in hopes of clarifying the status of __move_post_blt. If it's dead, quite a few changes would need to be made.

**Comment by MoonlightSentinel** [comment]
Ping, `move` should not have different in druntime and Phobos (regardless of the state of `opPostMove`)

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**